src/hotspot/share/opto/graphKit.cpp

```diff
@@ -3982,6 +3982,7 @@ Node* GraphKit::new_array(Node* klass_node,     // array klass (maybe variable)
     Node* ccast = alloc->make_ideal_length(ary_type, &_gvn);
     if (ccast != length) {
       _gvn.set_type_bottom(ccast);
+      ccast = _gvn.transform(ccast);
       record_for_igvn(ccast);
       replace_in_map(length, ccast);
```

Comment:
_gvn.set_type_bottom(ccast);
is not needed then. 
record_for_igvn(ccast);
is likely not needed either. No sure the reason there was no call to transform() in the first place.

Reply:
Thanks Roland! These codes are stble for a long time.

"_gvn.set_type_bottom(ccast);" is not needed as type will be set in "_gvn.transform".
But not sure if it is safe to remove "record_for_igvn".  Could you share more thoughts on it?  

```
eee15b163 hotspot/src/share/vm/opto/graphKit.cpp (Chuck Rasbold           2008-09-17 08:29:17 -0700 3984)       _gvn.set_type_bottom(ccast);
eee15b163 hotspot/src/share/vm/opto/graphKit.cpp (Chuck Rasbold           2008-09-17 08:29:17 -0700 3986)       record_for_igvn(ccast);
```

Reply:
FYI.  Tier1/2/3 release/fastdebug build pass when removing "_gvn.set_type_bottom(ccast);" and "record_for_igvn(ccast);"

Reply:
@rwestrel 
for "record_for_igvn" is likely not needed either.  Please help correct my understanding.

In GraphKit::new_array, CastIINode's _range_check_dependency is false and _gvn is not igvn. 
When "_gvn.transform(ccast)" calls CastIINode::Ideal, it either already in for_igvn (goes to "phase->record_for_igvn(this)") or in "post_loop_opts_igvn". 

No need call record_for_igvn again after " _gvn.transform(ccast)".

Reply:
The usual pattern at parse time is to call gvn.transform() and that's it. Later on, during igvn, if an input of a node changes, it's enqueued to be processed for igvn so at parse time, usually, there's no need to call record_for_igvn().
In this case, I think there's a call to record_for_igvn() because there's no call to gvn.transform(). The node has to be transformed at some point. If it's not done during gvn, it has to be during igvn and the cal to record_for_igvn() guarantees that.

Reply:
Thanks! New commit is added to remove not needed ops.